cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/SearchController.java

```diff
@@ -0,0 +1,63 @@
+package edu.asu.diging.cord19.explorer.web;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.cord19.explorer.core.model.impl.PublicationImpl;
+import edu.asu.diging.cord19.explorer.core.mongo.AffiliationSearchProvider;
+import edu.asu.diging.cord19.explorer.core.mongo.PublicationSearchProvider;
+import edu.asu.diging.cord19.explorer.core.mongo.impl.AffiliationPaperAggregationOutput;
+import edu.asu.diging.cord19.explorer.web.model.SearchType;
+
+@Controller
+public class SearchController {
+
+    
+    @Autowired
+    private AffiliationSearchProvider affSearchProvider;
+    
+    @Autowired
+    private PublicationSearchProvider pubSearchProvider;
+    
+    
+    
+    
+    @RequestMapping(value = "/search")
+    public String search(@RequestParam("searchType") String searchType,@RequestParam("search") String query, Model model,
+            @PageableDefault(size = 20) Pageable pageable) {
+        
+        if (searchType.equals(SearchType.AFFILIATIONS.toString())) {
+            List<AffiliationPaperAggregationOutput> matchedAffiliationsList = affSearchProvider.getRequestedPage(query, (long)pageable.getPageNumber(),
+                    pageable.getPageSize());
+            long affCount = affSearchProvider.searchResultSize(query);
+            model.addAttribute("matchedAffiliationsPage", matchedAffiliationsList);
+            addAttributes(model, pageable, query, affCount);
+            return "affResults";
+            
+            
+        } else  {
+            List<PublicationImpl> matchedPublicationsList = pubSearchProvider.getRequestedPage(query, (long)pageable.getPageNumber(),
+                    pageable.getPageSize());
+            long pubCount = pubSearchProvider.searchResultSize(query);
+            addAttributes(model, pageable, query, pubCount);
+            model.addAttribute("matchedPublicationsPage", matchedPublicationsList);
+            return "results";
+            
+        }
+    }
```

Comment:
thinking about this, to make this either to extend and maintain (since we will probably want to add more searches later):
- create a general searchprovider interface that has three methods:
   - getSearchType
   - getTotalResults (or something like that)
   - search
- each search provider implements that interface
- the getSearchType method returns the SearchType enum that they support
- the other two methods do the searching/total result counting
- add a searchproviderregistry that loads all service that implement the search provider interface into a map (searchtype -> serachprovider)
- then in this method just get the searchprovider for the passed search type and fill the model with the search results
- maybe have a map that maps search types to view names to return the appropriate view